src/tools/rados/rados.cc

```diff
@@ -1967,11 +1969,20 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
     }
   }
 
+  i = opts.find("pgid");
+  boost::optional<pg_t> pgid(i != opts.end(), pg_t());
```

Comment:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/rados/rados.cc:3646:1: error: jump to label ‘out’ [-fpermissive]
 out:
 ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/rados/rados.cc:1968:12: note:   from here
       goto out;
            ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/rados/rados.cc:1973:25: note:   crosses initialization of ‘boost::optional<pg_t> pgid’
   boost::optional<pg_t> pgid(i != opts.end(), pg_t());
```

Reply:
@tchaikov thanks Kefu. Needed to pull another commit. As `goto` statement was causing an issue.